File: skyramp/thrift-demo/sample-clients/local/python/addCart.py
from statistics import quantiles
from thrift import Thrift
from thrift.transport import THttpClient
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol

from thriftpy.demo import CartService
from thriftpy.demo.ttypes import CartItem

def main():
    uri = "cart-service-port50000.demo.skyramp.test/CartService"
    socket = THttpClient.THttpClient(f'http://{uri}')
    transport = TTransport.TBufferedTransport(socket)
    protocol = TBinaryProtocol.TBinaryProtocol(transport)
    transport.open()
    print("successfully connected to cart-service")
    client = CartService.Client(protocol)
    user_id = "abcde"
    quantity = 5
    cartItem = CartItem(product_id='OLJCESPC7Z', quantity=quantity)
    products = client.AddItem(user_id, cartItem )
    print(f"sucessfully added {quantity} products to cart for user {user_id}")

    user_id = "abcde"
    quantity = 5
    cartItem = CartItem(product_id='OLJCESPC7Z', quantity=quantity)
    cart = client.GetCart(user_id)
    print(f"sucessfully called GetCart user {user_id}. result {cart}")

    transport.close()

if __name__ == '__main__':
    try:
        main()
    except Thrift.TException as tx:
        print('%s' % tx)

File: skyramp/thrift-demo/sample-clients/local/golang/cmd/cart/addCart.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"os"

	api "sample-thrift/demo"
)

const (
	thriftHttpPath = "/CartService"
	localHost      = "127.0.0.1"
)

func main() {
	clientAddr := "thrift-demo.cart-service-port50000.checkout-system.skyramp.test"
	opt := NewDefaultOption()
	opt.HttpUrl = thriftHttpPath

	c, trans, err := NewThriftClient(clientAddr, opt)
	if err != nil {
		fmt.Printf("Failed to connect to server: %v", err)
		os.Exit(1)
	}
	err = trans.Open()
	if err != nil {
		fmt.Printf("Failed to connect to server: %v", err)
		os.Exit(1)
	}
	//
	// NOTE: Add Product to Cart
	//
	fmt.Println("connected to Cart Service")
	client := api.NewCartServiceClient(c)
	user_id := "abcde"
	product_id := "OLJCESPC7Z"
	fmt.Printf("Adding product %s to Cart\n", product_id)
	err = client.AddItem(context.Background(), user_id, &api.CartItem{ProductID: product_id, Quantity: 5})
	if err != nil {
		fmt.Printf("Failed to add item to cart: %v", err)
		os.Exit(1)
	}

	fmt.Printf("\nGet Cart\n")
	cart, err := client.GetCart(context.Background(), user_id)
	if err != nil {
		fmt.Printf("Failed to get cart: %v", err)
		os.Exit(1)
	}
	jsonProd, err := json.MarshalIndent(cart, "", "\t")
	if err != nil {
		fmt.Printf("Failed to Marshal the response from Cart Service: %v", err)
		os.Exit(1)
	}
	fmt.Println(string(jsonProd))
}
